presto-main/src/test/java/com/facebook/presto/cost/TestComparisonStatsCalculator.java

```diff
@@ -499,4 +520,97 @@ public void symbolToLiteralGreaterThanStats()
+                    .nullsFraction(0.1);
+        };
+
+        // Equal ranges
+        double rowCount = 2.7;
+        assertCalculate(new ComparisonExpression(EQUAL, new SymbolReference("u"), new SymbolReference("w")))
+                .outputRowsCount(rowCount)
+                .symbolStats("u", symbolAssert -> {
+                    symbolAssert.averageRowSize(8)
+                            .lowValue(0)
+                            .highValue(20)
+                            .distinctValuesCount(30)
+                            .nullsFraction(0);
+                })
+                .symbolStats("w", symbolAssert -> {
+                    symbolAssert.averageRowSize(8)
+                            .lowValue(0)
+                            .highValue(20)
+                            .distinctValuesCount(30)
+                            .nullsFraction(0);
+                })
+                .symbolStats("z", zUnchanged.apply(rowCount));
+
+        // One symbol's range is within the other's
+        rowCount = 4.6875;
+        assertCalculate(new ComparisonExpression(EQUAL, new SymbolReference("x"), new SymbolReference("y")))
+                .outputRowsCount(rowCount)
+                .symbolStats("x", symbolAssert -> {
+                    symbolAssert.averageRowSize(4)
+                            .lowValue(0)
+                            .highValue(5)
+                            .distinctValuesCount(10)
+                            .nullsFraction(0);
+                })
+                .symbolStats("y", symbolAssert -> {
+                    symbolAssert.averageRowSize(4)
+                            .lowValue(0)
+                            .highValue(5)
+                            .distinctValuesCount(10)
+                            .nullsFraction(0);
+                })
+                .symbolStats("z", zUnchanged.apply(rowCount));
+
+        // None of the ranges is included in the other
+        rowCount = 8.4375;
+        assertCalculate(new ComparisonExpression(EQUAL, new SymbolReference("x"), new SymbolReference("w")))
+                .outputRowsCount(rowCount)
+                .symbolStats("x", symbolAssert -> {
+                    symbolAssert.averageRowSize(4)
+                            .lowValue(0)
+                            .highValue(10)
+                            .distinctValuesCount(15)
+                            .nullsFraction(0);
+                })
+                .symbolStats("w", symbolAssert -> {
+                    symbolAssert.averageRowSize(8)
+                            .lowValue(0)
+                            .highValue(10)
+                            .distinctValuesCount(15)
+                            .nullsFraction(0);
+                })
+                .symbolStats("z", zUnchanged.apply(rowCount));
+
+        // None of the ranges is included in the other, and one symbol has much higher cardinality, so that has more NDV in intersect than the other in total
```

Comment:
None of the ranges is included in the other, and one symbol has much higher cardinality, so that it has bigger NDV in intersect than the other in total